### DISCUSSION TITLE
Invoices in the background

### DISCUSSION THREAD

**Discussion by leandro21b** [ACINQ/phoenix#362] [Improving Phoenix]
An interesting feature would be for the wallet to run in the background when waiting for a payment without the need for the wallet to be open to receive it.

**Comment by dpad85** [comment]
This feature is already supported: when Phoenix is closed/in the background, our node will pause any incoming payments and send a silent notification through FCM/APS to wake-up Phoenix, so that it connects and receives the payment.

This mechanism is not foolproof and still requires Phoenix to actually connect to receive the payment. Some Android phones are known to aggressively block background processes for "battery optimization" purposes. iOS is also quite restrictive and that can cause some issues. But in many/most cases, it works.
